pages/dataset/[datasetid].js

```diff
@@ -63,286 +77,309 @@ const Dataset = ({ dataset }) => {
+            About this Dataset
+          </h2>
+          {/* <a href="#">
+            <img src="/share.svg" alt="next" />
+          </a> */}
+        </div>
+        <div className="mb-20 font-karla text-lg">{dataset.description}</div>
+        <h2 className="mb-10 font-lato font-bold text-xl">File Preview</h2>
         <div className="mb-10">
-          {data && data.length ? (
+          {data && data.length != 0 ? (
             <CustomTable data={data} columns={columns} />
           ) : (
-            'No preview is available for this dataset.'
+            'NO PREVIEW FOR THIS DATASET'
           )}
         </div>
         <h2 className="mb-10 font-lato font-bold text-xl">Download</h2>
 
-        <div className="overflow-x-auto mb-10"> 
-          {Object.keys(dataset).includes('resources') && dataset.resources.length 
-            ? (
-              <table className="table-auto text-left text-sm mb-10 sm:mb-20 sm:text-base">
-                <thead className="bg-portal3">
-                  <tr>
-                    <th className="border border-black border-opacity-50 p-1 sm:p-4">
+        <div className="overflow-x-auto">
+          <table className="table-auto text-left text-sm mb-10 sm:mb-20 sm:text-base">
+            <thead className="bg-portal3">
+              <tr>
+                <th className="border border-black border-opacity-50 p-1 sm:p-4">
                   File size (MB)
-                    </th>
-                    <th className="border border-black border-opacity-50 p-1 sm:p-4">
+                </th>
+                <th className="border border-black border-opacity-50 p-1 sm:p-4">
                   File name
-                    </th>
-                  </tr>
-                </thead>
-                <tbody>
-                  {dataset.resources.map((resource, index) => {
-                    const filename = 'path' in resource
-                      ? `${resource.name}.${resource.path
-                        .split('.')
-                        .pop()}`
-                      : resource.name
-                    return (
-                      <>
-                        <tr key={index + '@resource'}>
-                          <td className="border border-black border-opacity-50 p-1 sm:p-4 lg:p-6">
-                            {(resource.bytes * 0.000001).toFixed(1)}
-                          </td>
-                          <td className="border border-black border-opacity-50 p-1 sm:p-4 lg:p-6">
-                            <a className="resource-download" href={`/api/dataset/${dataset.name}/files/default/${resource.path}`} download>
-                              {filename}
-                            </a>
-                          </td>
-                        </tr>
-                      </>
-                    )
-                  })
-                  }
-                </tbody>
-              </table>
-            )
-            : ('There is no resource available to download.')
```

Comment:
@risenW Do we need to double check whether we are showing the text `There is no resource available to download.` when there is no resource? It looks like we shouldn't see the dataset in the index page unless it has at least one resource, so I think it doesn't really matter, but if you can confirm this, it would be best :wink:.

Reply:
@estebanruseler For a dataset to be displayed in the homepage, it must have a datapackage.json. Now, some dataset can have a datapackage with no resource, so in that case I'm sure we should be able to display the dataset and notify the user that there's no resource for now. 

Do you want a different behaviour?

Reply:
@risenW the way you describe the scenario seems like what most users might expect:

* If the dataset is totally empty (no `datapackage.json`, possibly just a README or even nothing at all), there is no point in showing it in the index page.
* Whereas if it has a `datapackage.json` regardless of the number of resources it contains (or not), it should be shown since it could otherwise be confusing as people might see it on GitHub but not in the portal. Additionally, it's very likely that if a `datapackage.json` is present, it would have been created by the publisher and not the user, so we can think the dataset contained information (resources or metadata) at some point in its existence.

We can later document the fact that if they want to completely remove a dataset and not go back to it, they should just delete the repo from GitHub.

Reply:
@risenW pls note that Esteban and Sebastien, whilst having etymological similar roots, are different names :P